### DISCUSSION TITLE
What is this class, focus-visible:ring-ring what does it do?

### DISCUSSION THREAD

**Discussion by srinivasreddy** [shadcn-ui/ui#3458] [General]
What is this class `focus-visible:ring-ring` what does it do? I see it button.tsx, I understand the ring functionality. But I couldn't find any reference of it online? Was that a typo?

**Comment by flixlix** [comment]
This is what`ring-ring` does. It basically just sets the color of the ring to be used as the `--ring` custom css property defined in your css.

<img width="327" alt="image" src="https://github.com/shadcn-ui/ui/assets/61006057/88faff8c-eb81-4331-9ee2-6545139b91c6">

In these cases, extensions are your friend for checking the css behind a tailwind class.
https://tailwindcss.com/docs/editor-setup#intelli-sense-for-vs-code
